### DISCUSSION TITLE
Building Classification

### DISCUSSION THREAD

**Discussion by ghost** [r-lidar/lidR#533] [Q&A]
Hello Jean-Romain,

Is lidR capable of classifying buildings (Perhaps individually) or is this achieved another way? 

I would like to classify my original .laz into ground, buildings, and trees as separate data. Currently, I see the top surfaces of some buildings in my .laz file being classified as ground when I run `classify_ground(). `

Thank you!

![image](https://user-images.githubusercontent.com/97917300/154305479-ebe55a70-76d0-4078-94ae-2903d97d6d11.png)
Whole .laz plot. 

![image](https://user-images.githubusercontent.com/97917300/154305502-22ba2400-8d2b-4068-9b62-1ff2e943efed.png)
Ground classified plot.

**Comment by Jean-Romain** [comment]
There is no built in `classify_building()` strictly speaking but we do have flat surface recognition with `segment_shape()` that may be used for building segmentation. See section [13.2.1](https://r-lidar.github.io/lidRbook/pba.html#pba-applications-roof) of the book. However flat surface recognition is not building classification. It may work , at least partially. You can maybe design something on top of it.

Have you considered using another software for ground classification? Algorithm provided in lidR are designed for forested areas. Remember the description of the lidR package

> lidR: Airborne LiDAR Data Manipulation and Visualization for Forestry Applications

**Comment by Jean-Romain** [comment]
Another suggestion is to use a shapefile of the buildings if you have one and use `classify_poi()`

**Comment by ghost** [comment]
Thank you for getting back to me so quickly, it's very much appreciated. 

The project I'm working on looks to produce a 3D model from a LiDAR data set using any open-source package (R, Python based etc). 
Although angled towards forestry, lidR happens to be the best-documented package I've seen with the clearest instructions. 

I will have a go at running `segment_shape()` and importing a building shapefile and seeing where it gets me, it might be that part of this workflow takes part in another package (Perhaps QGIS). 

Thank you for your help!
